src/Client/Grpc/GrpcDurableTaskClient.cs

```diff
@@ -250,6 +290,11 @@ public override AsyncPageable<OrchestrationMetadata> GetAllInstancesAsync(Orches
     public override async Task<OrchestrationMetadata> WaitForInstanceStartAsync(
         string instanceId, bool getInputsAndOutputs = false, CancellationToken cancellation = default)
     {
+        if (this.options.SupportEntities)
```

Comment:
I think I also have to add the ` [CallerArgumentExpression("argument")] string? name = default` for the magic of compiler-generated argument descriptor to work.

Reply:
Since we need the same thing for `DurableTaskWorkerOptions` I am instead going to just pass in the `EnableEntitySupport` as a boolean argument to `Check.NotEntity`.